lightbeam/validate.py

```diff
@@ -18,19 +21,150 @@ def __init__(self, lightbeam=None):
     # Validates (selected) endpoints
     def validate(self):
         self.lightbeam.api.load_swagger_docs()
-        asyncio.run(self.lightbeam.api.load_descriptors_values())
+        validation_methods = self.lightbeam.config.get("validate",{}).get("methods",self.DEFAULT_VALIDATION_METHODS)
+        if type(validation_methods)==str and (validation_methods=="*" or validation_methods.lower()=='all'):
+            validation_methods = self.DEFAULT_VALIDATION_METHODS
+            validation_methods.append("references")
+        self.logger.info(f"validating by methods {validation_methods}...")
+        if "descriptors" in validation_methods:
+            # load remote descriptors
+            asyncio.run(self.lightbeam.api.load_descriptors_values())
+            self.lightbeam.reset_counters()
+            self.local_descriptors = self.load_local_descriptors()
+        
+        for endpoint in self.lightbeam.endpoints:
+            if "references" in validation_methods and "Descriptor" not in endpoint: # Descriptors have no references:
+                # We don't want every `do_validate_payload()` to separately have to open and scan
+                # local files looking for a matching payload; this pre-loads local data that
+                # might resolve references from within payloads of this endpoint.
+                # We assume that the data fits in memory; the largest Ed-Fi endpoints
+                # (studentSectionAssociations, studentSchoolAttendanceEvents, etc.) contain references
+                # to comparatively datasets (sections, schools, students).
+                self.load_local_reference_data(endpoint)
+                # create a structure which remote reference lookups can populate to prevent repeated lookups for the same thing
+                self.remote_reference_cache = {}
```

Comment:
This resets the cache for each endpoint, perhaps we only want to do it once outside the loop (at line 35) since multiple endpoints may reference the same (cacheable) resources?